.github/workflows/forge-build.yml

```diff
@@ -32,8 +32,19 @@ jobs:
       - name: "Install Foundry"
         uses: "foundry-rs/foundry-toolchain@v1"
 
+      - name:
+          "Cache the contracts and the node modules. If a match is found for the key, the cache is restored. Otherwise,
```

Comment:
Shouldn't it be after `FOUNDRY_PROFILE=test-optimized forge build` where it originally was?

Reply:
No. This is one of the things that I find super confusing about `actions/cache`.

the default action `actions/cache` performs BOTH the restore and the save.